coordinator/src/node/event_post_processing.rs

```diff
@@ -0,0 +1,47 @@
+use crate::db;
+use crate::node::Node;
+use crate::routing_fee::models::NewRoutingFee;
+use lightning::util::events::Event;
+use tokio::sync::watch;
+
+pub fn spawn(node: Node, mut node_event_receiver: watch::Receiver<Option<Event>>) {
+    tokio::spawn(async move {
+        loop {
+            match node_event_receiver.changed().await {
+                Ok(()) => {
+                    // we only handle event where the fee is set, i.e. events that do not concern a
```

Comment:
🤔 that comment seems off. Force closing has nothing to do with fees for payment routing.

Reply:
I can try to provide more context - my understanding is that `PaymentForwarded` is indeed influenced by channel force-close scenarios, see docs:

```rust
           PaymentForwarded {
		/// The incoming channel between the previous node and us. This is only `None` for events
		/// generated or serialized by versions prior to 0.0.107.
		prev_channel_id: Option<[u8; 32]>,
		/// The outgoing channel between the next node and us. This is only `None` for events
		/// generated or serialized by versions prior to 0.0.107.
		next_channel_id: Option<[u8; 32]>,
		/// The fee, in milli-satoshis, which was earned as a result of the payment.
		///
		/// Note that if we force-closed the channel over which we forwarded an HTLC while the HTLC
		/// was pending, the amount the next hop claimed will have been rounded down to the nearest
		/// whole satoshi. Thus, the fee calculated here may be higher than expected as we still
		/// claimed the full value in millisatoshis from the source. In this case,
		/// `claim_from_onchain_tx` will be set.
		///
		/// If the channel which sent us the payment has been force-closed, we will claim the funds
		/// via an on-chain transaction. In that case we do not yet know the on-chain transaction
		/// fees which we will spend and will instead set this to `None`. It is possible duplicate
		/// `PaymentForwarded` events are generated for the same payment iff `fee_earned_msat` is
		/// `None`.
		fee_earned_msat: Option<u64>,
		/// If this is `true`, the forwarded HTLC was claimed by our counterparty via an on-chain
		/// transaction.
		claim_from_onchain_tx: bool,
	},
```

I adapted the description slightly when refactoring the function; it's not in the doc-comment of `routing_fee::handle`. Feel free to make another suggestion if you have a more understandable description!

Reply:
Thanks for the explanation. 👍